### PR TITLE
Remove Phase-2 IT bricked design

### PR DESCRIPTION
**Pull request by tvami**
#### PR description:

* Removed `PixelBrickedDigitizerAlgorithm` since that design was discarded
* Removed `generic_position_formula_y_bricked` which seems to have been a remnant from the cleanup from the local reco
* since I was there I changed the `namespace` naming to obey rule 2.7
* Removed the bricked design from the topology in https://github.com/cms-sw/cmssw/pull/40443/commits/11e07be56362864c5e7dd877bb21039c48c36690

#### PR validation:

* Checked the dependencies if there is anything that wouldnt compile
* Ran wf `20834.0` to make sure the default Ph2 digitizer works

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport, and no backport needed

### CODE REVIEW

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by mmusich** [comment]
hold
   * has this been discussed in the phase-2 simulation meeting ? @emiglior . Is there any residual interest in this for phase-3 upgrades?

**Comment by tvami** [comment]
> has this been discussed in the phase-2 simulation meeting ? @emiglior 

It was discussed with Ernesto, I added you to the email thread

**Comment by mmusich** [comment]
> Perhaps I am misunderstanding your point: Do you mean remove all the occurrences of "isBricked()" (and similar) or also the include of the header files (e.g. PTrackerAdditionalParametersPerDetRcd.h) through which the brickedness was implemented?

yes, carrying around this additional complication for no apparent reason doesn't not seem very sustainable for maintenance reasons in the long run.

**Comment by tvami** [comment]
I can do another commit which will remove bricked from the topology, but only later today

**Comment by tvami** [comment]
So there is an `isBricked()` in the MTD
https://github.com/cms-sw/cmssw/blob/master/Geometry/MTDGeometryBuilder/interface/RectangularMTDTopology.h#L173
not really sure what to do with this, or if it has anything to do with the IT topology.

**Comment by mmusich** [comment]
> not really sure what to do with this, or if it has anything to do with the IT topology.

That's necessary because `RectangularMTDTopology` inherits from `PixelTopology` (which provides the interface). 
Once it's removed in the base class it can naturally be removed here as well. As you can see anyway always `return`s false.

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
I tried to fix the addOnTests in https://github.com/cms-sw/cmssw/pull/40443/commits/5a840b0197af0495679c61743aa65ddcc69e4364
but it still doesnt work... what else was I supposed to do?

File: HLTrigger/Configuration/test/OnLine_HLT_PIon.py [hunk 1]
```diff
@@ -3063,9 +3063,6 @@
   useMagVolumes = cms.bool( True ),
   returnTangentPlane = cms.bool( True )
 )
-process.TrackerAdditionalParametersPerDetESModule = cms.ESProducer( "TrackerAdditionalParametersPerDetESModule",
-  appendToDataLabel = cms.string( "" )
-)
```
Comment (Martin-Grunewald) [inline HLTrigger/Configuration/test/OnLine_HLT_PIon.py:3068]: Please remove the changes from HLTrigger/Configuration/test/OnLine_HLT_PIon.py and restore th eoriginal: this file is generated and the next iteration will overwrite any changes in here. Any changes should be done in HLTrigger/Configuration/python/customizeHLTforCMSSW.py

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py [hunk 2]
```diff
@@ -248,6 +248,11 @@ def customizeHLTfor40264(process):
 
     return process
 
+def customizeHLTfor40443(process):
+     for producer in esproducers_by_type(process, "TrackerAdditionalParametersPerDetESModule"):
+	delattr(producer, "TrackerAdditionalParametersPerDetESModule")
```
Comment (Martin-Grunewald) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py:253]: This should likely be something like:
```delattr(process,producer)```

Comment (Martin-Grunewald) [reply HLTrigger/Configuration/python/customizeHLTforCMSSW.py:253]: Try this:
```
labels = [producer.label_() for producer in esproducers_by_type(process, "TrackerAdditionalParametersPerDetESModule")]
for label in labels:
    delattr(process,label)
```

Comment (missirol) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py:253]: Please use the following or equivalent (and do not use tabs.. [*]).
```suggestion
     for foo in [bar for bar in esproducers_by_type(process, "TrackerAdditionalParametersPerDetESModule")]:
         delattr(process, foo.label())
```
[*]
```
Traceback (most recent call last):
[..]
    delattr(producer, "TrackerAdditionalParametersPerDetESModule")
TabError: inconsistent use of tabs and spaces in indentation
```

File: HLTrigger/Configuration/test/OnLine_HLT_PIon.py [hunk 3]
```diff
@@ -52,7 +52,7 @@
     test = cms.vstring( 'ECAL' ),
     emulator = cms.vstring( 'None' )
   ),
-  streamEventDisplay = cms.PSet( 
+ streamEventDisplay = cms.PSet(
```
Comment (missirol) [inline HLTrigger/Configuration/test/OnLine_HLT_PIon.py:55]: Picky, but please undo this change too (as mentioned, these files are only updated by HLT).

Comment (tvami) [reply HLTrigger/Configuration/test/OnLine_HLT_PIon.py:55]: ok done

**Comment by tvami** [comment]
@cmsbuild please test

**Comment by tvami** [comment]
Great, diffs in msg logger only, as expected!

**Comment by ChrisMisan** [comment]
+alca

- differences only in msgLogger
- 20834.0 passes

**Comment by missirol** [comment]
+hlt

**Comment by missirol** [comment]
Just for my own education: why is
https://github.com/cms-sw/cmssw/blob/8e41ed3be640843242b5f221694aab2edee35b1e/CondFormats/GeometryObjects/interface/PTrackerAdditionalParametersPerDet.h
not removed, or modified, as well? (it mentions 'brickedness')

**Comment by mmusich** [comment]
> Just for my own education: why is
https://github.com/cms-sw/cmssw/blob/8e41ed3be640843242b5f221694aab2edee35b1e/CondFormats/GeometryObjects/interface/PTrackerAdditionalParametersPerDet.h
not removed, or modified, as well? (it mentions 'brickedness')

Since we were anyway breaking a lot of code that depends on [how the tracker geometry is built](https://indico.cern.ch/event/1045483/contributions/4392063/attachments/2258285/3832430/Brickedness_SIM_AdeWit_0406.pdf), the layout of the class `PTrackerAdditionalParametersPerDet` was designed in such a way that it allows a "poor-man" schema evolution of the `CondFormat` (which is normally not possible), so that it can store additional parameters if needed in future without the need of introducing a new data format. For that reason, I would be in favor of keeping the class and the related record in `cmssw` in case we need to populate it with some other unforeseen feature - just in case - until the phase-2 Tracker layout is fully frozen.
References to the "brickedness" on the other hand could indeed be removed.

**Comment by mmusich** [comment]
unhold

**Comment by tvami** [comment]
My understanding is that `getBricked()` and `setBricked()` should not be removed bc of policy. `BRICKEDNESS = 0` could be removed, tho I'd prefer not to need to retrigger tests just for that single change.

**Comment by mmusich** [comment]
> My understanding is that getBricked() and setBricked() should not be removed bc of policy.

Why not? Accessors are not persisted and can freely come and go.

**Comment by tvami** [comment]
OK, I did it in the last push.

**Comment by tvami** [comment]
@cmsbuild ,  please test

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
I'll rebase, there is a conflict now for `customizeHLTforCMSSW`

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by mmusich** [comment]
@tvami there are still conflicts to resolve?

**Comment by tvami** [comment]
Yes, because in the meantime my own Charge Reweighting for Phase-2 PR got merged haha, I'll rebase again but maybe I'll wait for the next IB to come out, this PR is just too big to squash (which I would like to do)

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
+1
 * diffs in MsgLogger only

**Comment by civanch** [comment]
+1

**Comment by missirol** [comment]
Unfortunately, there is now a merge-conflict due to the HLT customisation, so this will have to be rebased.

**Comment by tvami** [comment]
> Unfortunately, there is now a merge-conflict due to the HLT customisation, so this will have to be rebased.

Yeah that's my 3rd rebase now... but I guess that happens when one changes 114 files under 8 signatures. 
Anyway, can we avoid testing again? It's obvious that https://github.com/cms-sw/cmssw/pull/40443/commits/3291941de5efe2d4f5e2171bc2acebd348c3c08c couldnt break anything. @missirol would you please sign for HLT and @civanch can you please resign?

**Comment by mmusich** [comment]
> It's obvious that https://github.com/cms-sw/cmssw/commit/3291941de5efe2d4f5e2171bc2acebd348c3c08c couldnt break anything.

are merges acceptable?

**Comment by missirol** [comment]
> but I guess that happens when one changes 114 files under 8 signatures.

I think the main reason is that the HLT customisations for PRs converge to one place, and that is prone to (simple) merge conflicts. A couple of PRs with HLT customisations happened to be merged before this one, for different reasons.

> It's obvious that..

However obvious, I would skip testing only if there was a very good reason, or urgency, which I don't see. Plus, I would rebase and squash, removing the merge commit.

**Comment by tvami** [comment]
ok fine, I'll rebase and squash.

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by francescobrivio** [comment]
+1

 - removal of bricked geometry is correct from `alca`/`db` point of view
 - tests (before rebasing) were fine

**Comment by civanch** [comment]
+1

**Comment by missirol** [comment]
+hlt

**Comment by missirol** [comment]
@cms-sw/reconstruction-l2 @cms-sw/upgrade-l2 

Can you please review this PR?

**Comment by AdrianoDee** [comment]
On it

**Comment by mmusich** [comment]
@smuzaffar cms-bot flagged this as `tracking` 

Screenshot from 2023-01-19 18-37-54

but I couldn't find a single package changed here matching with the list at https://github.com/cms-sw/cms-bot/blob/master/categories_map.py#L1768-L1796. 
Can you check if there is a bug in the list and / or in the matching code?

**Comment by mmusich** [comment]
type -tracking
   * assignment looks buggy

**Comment by smuzaffar** [comment]
@mmusich , `RecoTracker` at https://github.com/cms-sw/cms-bot/blob/master/categories_map.py#L1786  matched file `Configuration/Geometry/python/GeometryRecoTracker_cff.py` that is why bot added the `tracking` label :-)

**Comment by clacaputo** [comment]
+reconstruction

**Comment by mmusich** [comment]
> RecoTracker at https://github.com/cms-sw/cms-bot/blob/master/categories_map.py#L1786 matched file Configuration/Geometry/python/GeometryRecoTracker_cff.py that is why bot added the tracking label

@smuzaffar hmm, would `RecoTracker/` avoid that?

**Comment by smuzaffar** [comment]
> would RecoTracker/ avoid that?

yes

**Comment by AdrianoDee** [comment]
+upgrade

**Comment by AdrianoDee** [comment]
I was reading the thread: is it there at the end something that documents this decision?

**Comment by mmusich** [comment]
@emiglior 

> Meanwhile, I will cross check again with people working on ph-3 about their actual plans on bricked pixels.

Did phase-3 people get back to you about it?

**Comment by mmusich** [comment]
> is it there at the end something that documents this decision?

I guess it will be discussed today at the [ph2 tracker simulation meeting](https://indico.cern.ch/event/1242566/)

**Comment by AdrianoDee** [comment]
I see , thanks!

**Comment by emiglior** [comment]
> @emiglior
> 
> > Meanwhile, I will cross check again with people working on ph-3 about their actual plans on bricked pixels.
> 
> Did phase-3 people get back to you about it?

please go on with the removal

**Comment by tvami** [comment]
hi @cms-sw/orp-l2 can we merge this soon, before I'll need to rebase again? Thanks!

**Comment by perrotta** [comment]
+1
- Phase2 tracker approved it, see https://github.com/cms-sw/cmssw/pull/40443#issuecomment-1398270483
- The removals applied in quite several files did not break the builds, nor resulted in any visible difference in the test outputs

**Comment by perrotta** [comment]
@tvami a unit test failure in the IBs shows that there is a remaining inclusion of TrackerAdditionalParametersPerDet_cfi.py in DQM/SiStripCommon/test/testTkHistoMap_cfg.py: could you please have a look?

**Comment by tvami** [comment]
ok, I'm on it
